### ISSUE TITLE
What `opt_levels` from APEX does accelerate's FP16 correspond to?

### ISSUE THREAD

**Issue by phiyodr** [huggingface/accelerate#912]
Dear accelerate team,

what `opt_levels` from [APEX](https://nvidia.github.io/apex/amp.html) does accelerate's FP16 correspond to?

* O1: Mixed Precision 
* O2:  “Almost FP16” Mixed Precision
* O3: FP16 training

(also if you use `torch.amp` and not `apex.amp`)
I'm asking since the key below is called `mixed_precision` (which equals O1 and O2) and the value means O3. And somehow this confuses me a bit.

```bash
>>cat default_config.yaml 
compute_environment: LOCAL_MACHINE
deepspeed_config: {}
distributed_type: 'NO'
downcast_bf16: 'no'
fsdp_config: {}
gpu_ids: '0'
machine_rank: 0
main_process_ip: null
main_process_port: null
main_training_function: main
mixed_precision: fp16          <--------------
num_machines: 1
num_processes: 1
rdzv_backend: static
same_network: true
use_cpu: false
```

I think it means FP16 only (O3). Am I right?

Thanks in advance!

**Comment by phiyodr** [comment]
So it's FP16 only?

**Comment by pacman100** [comment]
Hello @phiyodr, please go through the discussion here: https://discuss.pytorch.org/t/torch-cuda-amp-how-to-set-optimization-level/92744

Tldr: `PyTorch Native AMP is similar to the recommended O1 level and doesn’t use any other opt_levels`

**Comment by phiyodr** [comment]
Thank you! And thanks @pacman100 for the link! That was the one information I was searching for ;)
